source/minio-cli/minio-kes-reference.rst

```diff
@@ -0,0 +1,421 @@
+======================================
+MinIO Key Encryption Service (``kes``)
+======================================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 2
+
+.. mc:: kes
+
+Overview
+--------
+
+The MinIO Key Encryption Service (KES) is a stateless and distributed
+key-management system for high-performance applications. KES provides
+a bridge between applications running in bare-metal or orchestrated
+environments to centralised KMS solutions. 
+
+This page provides a reference for the :mc:`kes` command line tool. For more
+complete conceptual information on KES, see :ref:`minio-kes`.
+
+For tutorials on deploying KES to support Server-Side Object Encryption, see:
+
+- Server-Side Object Encryption with Thales CipherText
+- Other similar tutorials.
+
+.. _minio-kes-installation:
+
+Installation
+------------
+
+.. include:: /includes/minio-kes-installation.rst
+
+.. _minio-kes-config:
+
+KES Configuration File
+----------------------
+
+The KES configuration file is a ``YAML`` document containing information for
+running the KES server. Use 
+:mc-cmd-option:`kes server config` to specify the path to the configuration file 
+on startup. 
+
+General Configuration
+~~~~~~~~~~~~~~~~~~~~~
+
+.. kesconf:: address
+
+   The network addresses and port on which :mc:`kes server` binds to on startup.
+   Specify a string as ``IP:PORT``:
+
+   - To specify only a custom IP address, omit ``:PORT``.
+
+   - To specify only a custom port, omit ``IP``.
+
+   Defaults to port ``7373`` on all network addresses on the host machine.
+
+.. kesconf:: root
+
+   Specify ``disabled`` to disable the :ref:`root <minio-kes-iam-root>` identity
+   on the KES server. The ``root`` identity has super administrator access to
+   the KES server.
+
+   If omitted, the :mc:`kes server` *requires* specifying an x.509 certificate
+   using the :mc-cmd-option:`~kes server root` command line option.
+
+TLS Configuration
+~~~~~~~~~~~~~~~~~
+
+The following section describes TLS-related configuration for the KES server:
+
+.. code-block:: yaml
+
+   tls:
+      key: <string>
+      cert: <string>
+
+.. kesconf:: tls
+
+   The root field for TLS-related configuration.
+
+.. kesconf:: tls.key
+
+   The path to the TLS private ``.key`` file KES uses when establishing
+   TLS-secured communications.
+
+   Ensure the Certificate Authority used to sign the private key exists in the
```

Comment:
Private keys are not signed. In general, there is a private key - which, like the name indicates, **nobody** else should know. Then there is a public key that corresponds to the private key. The public key, like the name indicates, can be known by anyone. So, there is a public-private-key pair.

Last but not least there is a certificate. The certificate contains the public key - **not** the private key. The certificate describes: Who ones the public key - and the corresponding private key. The certificate is signed by an authority. With this signature the authority makes a publicly verifiable statement:
*I, the `authority`, have verified that the following public key `public-key` actually belongs to `person/server/system/...`*